get.go

```diff
@@ -300,7 +300,7 @@ func (s Style) GetBorderTopWidth() int {
 // runes of varying widths, the widest rune is returned. If no border exists on
 // the top edge, 0 is returned.
 func (s Style) GetBorderTopSize() int {
-	if !s.getAsBool(borderTopKey, false) {
+	if !s.getAsBool(borderTopKey, s.shouldAutoEnableBorder()) {
```

Comment:
I think this should probably either short-circuit with `s.shouldAutoEnableBorder` or continue to have `false` as the second arg here. I don't think this function call is necessary unless I'm missing something :thinking: 

e.g. 
```go
if !s.shouldAutoEnableBorder() || !s.getAsBool(borderTopKey, false)
```

Although, again not sure this check saves us in efficiency

Reply:
If it's to account for the border value set in `BorderStyle`, wouldn't that be handled already at line 306?

Reply:
The double negative `!a || !b` style is confusing (and I think it should actually be `&&` not `||`). However I think if we invert the `if` condition it will be easier to follow:

```go
if s.shouldAutoEnableBorder() || s.getAsBool(borderTopKey, false) {
	return s.getBorderStyle().GetTopSize()
}
return 0
```

It breaks the Go idiom of putting the happy path return at the bottom, but I think the improvement in clarity is worth it.

Reply:
> If it's to account for the border value set in `BorderStyle`, wouldn't that be handled already at line 306?

That would ignore the show/hide border flags set on the style, e.g. under `borderTopKey`, which would be a regression I believe. At the very least it would change the behavior of many existing programs which users could rightfully consider a regression.